src/version_specifier.cpp

```diff
@@ -111,9 +111,9 @@ auto wlcs::LessThanVersion::select_version(
     {
         return {};
     }
-    else if (lesser_version <= max_available_version)
+    else if (max_allowed_version <= max_available_version)
     {
-        return {std::min(lesser_version, max_supported_version)};
+        return {std::min(max_allowed_version, max_supported_version)};
```

Comment:
Likewise, I think this should be returning the maximum supported version less than the specified version. So, if the compositor supports v5, `LessThanVersion{8}.select_version` should return `{5}`. (This is not critical, since always selecting v1 would *technically* match the behaviour of `LessThanVersion` :wink: )